### ISSUE TITLE
mirakuru.exceptions.TimeoutExpired

### ISSUE THREAD

**Issue by hugh-nguyen** [ClearcodeHQ/pytest-mysql#386]
### What action do you want to perform
Hi I am trying to run pytest-mysql in a docker image which looks like this
```
FROM amazonlinux:2

COPY docker/ca-bundle.crt /etc/pki/ca-trust/source/anchors/
RUN update-ca-trust

RUN yum install -y python37 wget

RUN rpm -Uvh https://dev.mysql.com/get/mysql57-community-release-el7-11.noarch.rpm
RUN rpm --import https://repo.mysql.com/RPM-GPG-KEY-mysql-2022
RUN yum install -y mysql-community-server libmysqlclient-dev mysql-devel gcc gcc-c++ python-devel python3-devel unixODBC-devel

COPY pipeline/files/datacollector/requirements.txt /
COPY testing/testing-requirements.txt /
RUN python3.7 -V
RUN pip3.7 install -r /requirements.txt -r /testing-requirements.txt
```

### What are the results
I keep getting this mirakuru.exceptions.TimeoutExpired error. Below is a log.
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.7/site-packages/pytest_mysql/factories/process.py:135: in mysql_proc_fixture
    with mysql_executor:
/usr/local/lib/python3.7/site-packages/mirakuru/base.py:181: in __enter__
    return self.start()
/usr/local/lib/python3.7/site-packages/pytest_mysql/executor.py:150: in start
    super().start()
/usr/local/lib/python3.7/site-packages/mirakuru/base.py:525: in start
    self.wait_for(self.check_subprocess)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pytest_mysql.executor.MySQLExecutor: "mysqld_saf..." 0x7f9251730650>
wait_for = <bound method Executor.check_subprocess of <pytest_mysql.executor.MySQLExecutor: "mysqld_saf..." 0x7f9251730650>>

    def wait_for(
        self: SimpleExecutorType, wait_for: Callable[[], bool]
    ) -> SimpleExecutorType:
        """
        Wait for callback to return True.
    
        Simply returns if wait_for condition has been met,
        raises TimeoutExpired otherwise and kills the process.
    
        :param callback wait_for: callback to call
        :raises: mirakuru.exceptions.TimeoutExpired
        :returns: itself
        :rtype: SimpleExecutor
        """
        while self.check_timeout():
            if wait_for():
                return self
            time.sleep(self._sleep)
    
        self.kill()
>       raise TimeoutExpired(self, timeout=self._timeout)
E       mirakuru.exceptions.TimeoutExpired: Executor <pytest_mysql.executor.MySQLExecutor: "mysqld_safe --datadir=/tmp/pytest-of-root/pytest-0/pytest-mysql-mysql_my_proc0/mysqldata_23386 --pid-file=/tmp/pytest-of-root/pytest-0/pytest-mysql-mysql_my_proc0/mysql-server.23386.pid --port=23386 --socket=/tmp/pytest-of-root/pytest-0/pytest-mysql-mysql_my_proc0/mysql.23386.sock --log-error=/tmp/pytest-of-root/pytest-0/pytest-mysql-mysql_my_proc0/mysql-server.-1.log --tmpdir=/tmp/pytest-of-root/pytest-0/pytest-mysql-mysql_my_proc0 --skip-syslog " 0x7f9251730650> timed out after 60 seconds

/usr/local/lib/python3.7/site-packages/mirakuru/base.py:450: TimeoutExpired
```

I did the same thing on Debian and Ubuntu and I eventually always get this error, but it doesn't happen on my host linux computer.

Please any help diagnosing this issue would be really appreciated as I've been working really long hours for a week to try to get this to work on docker and it always ends with this error.

Thanks
Hugh

**Comment by fizyk** [comment]
Sorry it took me this long, if you could have working example of code (some simple test, config and the docer) that would be helpful.

(Docker file, minimum test, config and the way you run your tests.
Alternatively, you could check the mysql logfiles

**Comment by xmikew** [comment]
I was getting the same issue. It turned out I had quoted the mysql_host in my pytest.ini:

**Broken ini**
```
mysql_host = "127.0.0.1"
mysql_dbname = "mydb"
```
This got passed into the executor as "'127.0.0.1'" which broke the TCPExecutor piece although I could pause the test and connect to the db just fine from command-line after the db started.

Removed the quotes:
```
mysql_host = 127.0.0.1
mysql_dbname = mydb
```
and things started working.

**Comment by xmikew** [comment]
Generally docker containers are running commands as root, and when the /tmp directories are created by pytest-mysql they are created as root. While mysqld --initialize-insecure can now take a --user parameter, pytest-mysql does not pass it and even if it did the temporary directory structure up to the datadir is all owned by root, rwx------  (iirc). So the initialize works, and if you don't specify mysql_user as a non-root and you are running pytest-mysql >= 2.5.0 then you're probably fine.

If you are running pytest-mysql < 2.5.0 or you specify a mysql_root user as non-root (e.g. mysql) then you'll get this error.
Before 2.5.0, the --user parameter is not passed to mysqld_safe startup. If the user calling mysqld_safe is root (and not specified on the command-line) it will default to user "mysql" and now it can't read any of the /tmp files (owned by root), fails to start and then TCPExecutor just waits for the process to start even though it's already died.

***TL;DR***

**`requirements_test.txt`**
pytest-mysql>=2.5.0

**`pytest.ini`**
mysql_user = root
